src/main/java/uk/gov/hmcts/reform/sscs/service/ServiceHearingsService.java

```diff
@@ -57,10 +67,37 @@ public ServiceHearingValues getServiceHearingValues(ServiceHearingRequest reques
     public List<ServiceLinkedCases> getServiceLinkedCases(ServiceHearingRequest request)
         throws GetCaseException {
 
-        SscsCaseData caseData = ccdCaseService.getCaseDetails(request.getCaseId()).getData();
+        String caseId = request.getCaseId();
+        List<SscsCaseDetails> mainCaseData = ccdCaseService.getCasesViaElastic(List.of(request.getCaseId()));
+
+        if (mainCaseData == null || mainCaseData.size() != NUM_CASES_EXPECTED) {
+            throw new IllegalStateException(
+                "Invalid search data returned: one case is required. Attempted to fetch data for " + caseId);
+        }
+
+        SscsCaseData caseData = mainCaseData.get(0).getData();
+
+        List<String> linkedReferences = Optional.ofNullable(caseData.getLinkedCase())
+            .orElseGet(Collections::emptyList).stream()
+            .filter(Objects::nonNull)
+            .map(CaseLink::getValue)
+            .filter(Objects::nonNull)
+            .map(CaseLinkDetails::getCaseReference)
+            .collect(Collectors.toList());
+
+        log.info("{} linked case references found for case: {}", linkedReferences.size(), caseData.getCaseReference());
+
+        List<SscsCaseDetails> linkedCases = ccdCaseService.getCasesViaElastic(linkedReferences);
```

Comment:
Just a suggestion, we can make a return instantly from this line, using stream().map(it -> buildServiceLinkedCases)